scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -248,29 +248,32 @@ def json_report(self, filename, version="NA"):
             handler_log = os.path.join(instance.build_dir, "handler.log")
             build_log = os.path.join(instance.build_dir, "build.log")
             device_log = os.path.join(instance.build_dir, "device.log")
-
             handler_time = instance.metrics.get('handler_time', 0)
             ram_size = instance.metrics.get ("ram_size", 0)
             rom_size  = instance.metrics.get("rom_size",0)
+            percentage_usage_ram = instance.metrics.get("percentage_usage_ram", 0)
```

Comment:
I think getting information about the percentage of memory usage is necessary. The build.log file contains information about the rate RAM/FLASH usage calculated based on the real available memory. Therefore, it is no need to calculate memory usage additionally. Given this fact, I consider the addition of information on available memory to the report redundant

Reply:
@nashif I am not sure if we are on the same page. This is an example of information available in the build.log. 
``` 
[144/144] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       61588 B         1 MB      5.87%
             RAM:          9 KB       256 KB      3.52%
        IDT_LIST:          0 GB         2 KB      0.00%
  ```
 By available memory, you mean the `Region size` or memory left, i.e. `region size - used size`? 
I think memory left can be useful, since it can help with understanding how close to the edge we are, in the terms of absolute values (kB left). Not sure why Region Size would be useful to store in the report for every image, can you elaborate if this is what you prefer?

>  percentage can then be calculated from used and available if needed.

We want to have the percentage in the report, since it was already calculated and prefer not to have to recalculate it again, after the report is ready. Such json report can then be directly feed to the data base (elastic search, grafana, etc). We found monitoring of the percentage useful and are planning to have an internal triggers which will warn us when % raise above a given value for our samples and applications.

Reply:
@nashif thanks for the clarification. Code has been changed.